### PR TITLE
SS: Dash Segments Videos are not supported by OpenCV VideoCapture.

### PR DESCRIPTION
**Pull request by DynamiteC**
## Brief Description
# Ignoring a Video Protocol not supported by OpenCV

Just continuing the loop when stream protocol is of http_dash_segments.
`                
if stream_protocol == 'http_dash_segments':
                    continue
`

### Requirements / Checklist

- [x] I have read the [PR Guidelines](https://abhitronix.github.io/vidgear/latest/contribution/PR/#submitting-pull-requestpr-guidelines).
- [x] I have read the [Documentation](https://abhitronix.github.io/vidgear/latest).
- [ ] I have updated the documentation accordingly(if required).

### Context

I was working on reading frames from youtube when the source URL sometimes failed so I checked, and found that when it failed was solely due to the URL was not supported by OpenCV so resolved that by just continuing the loop in case of HTTP DASH Segments Video

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Screenshots (if available):

Screenshot from 2022-06-16 12-14-03

### CODE REVIEW

**Comment by abhiTronix** [comment]
@DynamiteC Thanks for the PR, this is a long pending issue. I'll test it locally and let you know.

**Comment by DynamiteC** [comment]
Okay will do, sorry for that.

**Comment by DynamiteC** [comment]
Is this correct now?

**Comment by DynamiteC** [comment]
Now updated, didnt changed from below options. sorry for so many commits.

**Comment by DynamiteC** [comment]
> @DynamiteC Ok leave it, just a bit hard to see changes. No worries I'll change it after merge(if accepted).

Alright!

File: vidgear/gears/camgear.py [hunk 1]
```diff
@@ -1,544 +1,546 @@
...
+            # extract streams
+            streams = {}
+            streams_copy = {}
+            for stream in self.streams_metadata:
+                # extract useable metadata
+                stream_dim = stream.get("resolution", "")
+                stream_url = stream.get("url", "")
+                stream_protocol = stream.get("protocol", "")
+                if stream_protocol == 'http_dash_segments':
+                    continue
```
Comment (abhiTronix) [inline vidgear/gears/camgear.py:162]: @DynamiteC Remove this code:
```py
                if stream_protocol == 'http_dash_segments':
                    continue
```

File: vidgear/gears/camgear.py [hunk 2]
```diff
@@ -1,544 +1,546 @@
...
+                stream_with_video = (
+                    False if stream.get("vcodec", "none") == "none" else True
+                )
+                stream_with_audio = (
+                    False if stream.get("acodec", "none") == "none" else True
+                )
+                # streams must contain video
+                if stream_with_video and stream_dim and stream_url:
```
Comment (abhiTronix) [inline vidgear/gears/camgear.py:170]: @DynamiteC Put the code here by changing following logic:

```py
                if stream_with_video and stream_dim and stream_url:
```

to

```py
                if stream_with_video and stream_dim and stream_url and stream_protocol != "http_dash_segments":
```

**Review by abhiTronix — CHANGES_REQUESTED** [review]
@DynamiteC Remove your code and place it in the required `if` logic.

**Comment by abhiTronix** [comment]
@DynamiteC Thanks, LGTM!

**Review by abhiTronix — APPROVED** [review]
Changes are approved. I'll merge after CI tests are completed.

**Comment by DynamiteC** [comment]
Alright thanks, i forgot that you follow double quotes so I had to make another comit for that. thank you!!

**Comment by abhiTronix** [comment]
> Alright thanks, i forgot that you follow double quotes so I had to make another comit for that. thank you!!

Generally you must avoid single quotes for text in Python, as they're used for anything that behaves like an Identifier.

**Comment by DynamiteC** [comment]
Thank you, I will keep that in mind. I still learning python. If you have any recommendations on where to get advanced on python then please do. Thanks a lot!

**Pull request merged**